contracts/MemeFactory.sol

```diff
@@ -419,26 +412,32 @@ abstract contract MemeFactory {
     }
 
     /// @dev Converts collected reference token to OLAS.
-    function scheduleOLASForAscendance(uint256 slippage) external virtual {
+    function scheduleOLASForAscendance(uint256 amount, uint256 slippage) external virtual {
         require(_locked == 1, "Reentrancy guard");
         _locked = 2;
 
         // Slippage limit requirement
         require(slippage <= maxSlippage);
 
-        uint256 localAscendance = scheduledForAscendance;
-        require(localAscendance > 0, "Nothing to burn");
+        if (amount > scheduledForAscendance) {
+            amount = scheduledForAscendance;
+        }
+        require(amount > 0, "Nothing to burn");
 
+        // shouldn't this be inside _buyOLAS as its chain specific?
```

Comment:
@kupermind I think we should move the oracle logic into _buyOlas and call it from there. It's logically coupled.

Reply:
OK - solved differently now.